### PR TITLE
Added documentation for readenvfile() (CFE-2650)

### PR DESCRIPTION
**Pull request by olehermanse**
Example does not work yet, see: https://tracker.mender.io/browse/CFE-2656

### CODE REVIEW

**Comment by olehermanse** [comment]
@nickanderson @jimis @kacf Please review.

File: reference/functions/readdata.markdown [hunk 1]
```diff
@@ -41,6 +45,6 @@ Output:
 
 [%CFEngine_include_snippet(readdata.cf, #\+begin_src\s+example_output\s*, .*end_src)%]
 
-**See also:** `readcsv()`, `readyaml()`, `readjson()`, and `data` documentation.
+**See also:** `readcsv()`, `readyaml()`, `readjson()`, `readenvfile()`, `readenand `data` documentation.
```
Comment (nickanderson) [inline reference/functions/readdata.markdown]: Typo? readendand `data`?

File: reference/functions/readenvfile.markdown [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+**Example:**
+
+```cf3
+    vars:
+      "loadthis"
+         data => readenvfile("/etc/os-release", 4000);
+```
+
+Complete example, run with `cf-agent -f file.cf`:
```
Comment (nickanderson) [inline reference/functions/readenvfile.markdown]: I would simply include the example from core/examples.

For example: https://github.com/cfengine/documentation/blob/master/examples/example-snippets/general.markdown#hello-world and https://docs.cfengine.com/docs/3.10/examples-example-snippets-general.html#hello-world

https://github.com/cfengine/documentation/blob/master/cheatsheet.markdown#symlink-example
https://docs.cfengine.com/docs/master/markdown-cheatsheet.html#symlink-example

File: reference/functions/readenvfile.markdown [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        "jsonstring"
+            string =>  storejson(envdata);
+    # Create / modify json file with jsonstring
+    files:
+        "/tmp/os-release.json"
+            create => "true",
+            edit_line => insert_lines("$(jsonstring)"),
+            edit_defaults => empty;
+    reports:
+        "/etc/os-release converted to /tmp/os-release.json";
```
Comment (nickanderson) [inline reference/functions/readenvfile.markdown]: Want to show the content of that file?

```
    reports:
￼        "/etc/os-release converted to /tmp/os-release.json"
          printfile => cat("/tmp/os-release.json");
```

File: reference/functions/readdata.markdown (hunk 1 above)
Comment (olehermanse) [inline reference/functions/readdata.markdown]: Thanks, fixed.

File: reference/functions/readenvfile.markdown (hunk 3 above)
Comment (olehermanse) [inline reference/functions/readenvfile.markdown]: Thanks, added.

**Comment by olehermanse** [comment]
@nickanderson Thanks for the suggestions. I believe I fixed all of them. Can you, or perhaps @jimis review so we can get this merged?

**Comment by olehermanse** [comment]
Needs: https://github.com/cfengine/core/pull/2869

**Comment by nickanderson** [comment]
Great thanks!

**Pull request merged**